src/Apps/Legion/Contexts/GraphContext/GraphContext.types.ts

```diff
@@ -0,0 +1,55 @@
+import { ICustomGraphData } from '../../Components/GraphVisualizer/GraphTypes.types';
+
+export interface IGraphNode<T> {
+    /** color of the node */
+    color?: string;
+    /** data bag to attach to the node */
+    data: T;
```

Comment:
Perfect, this we we can pass the cooked data (in my current implementation ITwin objects) here to keep track of twin graph.